scope.go

```diff
@@ -261,118 +419,233 @@ func (s *scope) reportLoopRun() {
 
 // reports current registry with scope status lock held
 func (s *scope) reportRegistryWithLock() {
+	scopesToRemove := map[string]struct{}{}
 	s.registry.RLock()
 	if s.reporter != nil {
-		for _, ss := range s.registry.subscopes {
-			ss.report(s.reporter)
+		for name, ss := range s.registry.subscopes {
+			if ss.report(s.reporter) {
```

Comment:
the `report` function doesn't check whether all sub-scopes are expired, assuming that's because the `registry` tracks all scopes at the root level, they're not managed like a tree right?

Reply:
Yeah all scopes are tracked at root level.

Reply:
Yes, they're managed on a flat level.